src/scripts/main.js

```diff
@@ -1,3 +1,68 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('.field');
+  const appendRowButton = document.querySelector('.append-row');
+  const removeRowButton = document.querySelector('.remove-row');
+  const appendColumnButton = document.querySelector('.append-column');
+  const removeColumnButton = document.querySelector('.remove-column');
+
+  const MIN = 2;
+  const MAX = 10;
+
+  function updateButtons() {
+    const rowCount = table.rows.length;
+    const columnCount = table.rows[0].cells.length;
+
+    appendRowButton.disabled = rowCount >= MAX;
+    removeRowButton.disabled = rowCount <= MIN;
+    appendColumnButton.disabled = columnCount >= MAX;
+    removeColumnButton.disabled = columnCount <= MIN;
+  }
+
+  function appendRow() {
+    if (table.rows.length < MAX) {
+      const row = table.insertRow();
+
+      for (let i = 0; i < table.rows[0].cells.length; i++) {
+        row.insertCell();
+      }
+
+      updateButtons();
+    }
+  }
+
+  function removeRow() {
+    if (table.rows.length > MIN) {
+      table.deleteRow(-1);
+      updateButtons();
+    }
+  }
+
+  function appendColumn() {
+    if (table.rows[0].cells.length < MAX) {
```

Comment:
Similar to the row check, ensure that the table has at least one row before accessing `table.rows[0]` to avoid potential errors if the table is empty.